### DISCUSSION TITLE
Creating a new branch, then starting a codespace defaults to `main`

### DISCUSSION THREAD

**Discussion by jeffrafter** [community/community#5930] [Codespaces] [answered]
Hello Codespace friends! I've run into a user-experience that has gotten me in trouble a few times and thought I would call out my expectations:

Often I want to open a codespace for a branch. I create a new branch from the repository page on the site. Then I click the green code button to create a new codespace. That codespace opens right away and it defaulted to the `main` branch. I start working away only to realize later that I've made a mistake. Now I need to rebase/merge/something to get my branch to be mergeable. 

I'd love it if opening a new codespace defaulted to the branch you are currently on... or at least asked you if you are not on the the `default_branch`. Additionally, once in the codespace, the source control panel doesn't indicate the branch name well (it is in the placeholder for the commit message before you type):

![image](https://user-images.githubusercontent.com/4064/135656298-c3372723-daae-4c8e-8c19-7744d52b589f.png)

The GitHub Desktop interface is much more explicit here and prevents mistakes.

**Comment by matthewisabel** [comment, marked as answer]
Hey @jeffrafter! Is this the flow you're using:

https://user-images.githubusercontent.com/2132776/135918712-00cb180b-a07f-46f6-828f-ec451d593ae0.mp4

I end up on the test branch there from the web in my case. There was a bug here about 10 days ago that was fixed that was around for a day or two, but it sounds like you're still hitting something or possibly you have a different flow?

**Reply by jeffrafter** [reply]
I was definitely using that flow, and I think I was assuredly hitting the bug. Thanks!

**Reply by jeffrafter** [reply]
More recently I was running into alternative issues where opening the codespace didn't show anything in the lower status bar for branch. Looking in the terminal I could see it was now putting me on the expected branch... but it was unclear otherwise. Refreshing the codespace made the status bar item show up. 

The scenario was with a group of 8 people on zoom all launching new code spaces simultaneously. 50% of them didn't load correctly and needed to be refreshed. It almost felt like a race condition.
